src/ConnectionManager.ts

```diff
@@ -398,11 +399,10 @@ export class ConnectionManager {
         }
         await connection.onRemove?.();
         const connectionIndex = this.connections.indexOf(connection);
-        this.connections.splice(connectionIndex, 1);
-        if (this.getAllConnectionsForRoom(roomId).length === 0) {
-            log.info(`No more connections in ${roomId}, leaving room`);
-            await this.as.botIntent.leaveRoom(roomId);
+        if (connectionIndex === -1) {
+            throw Error('Could not find connection index');
         }
+        this.connections.splice(connectionIndex, 1);
```

Comment:
Pretty much at the moment. The problem we found was if someone used the widget to clear up their connections, the bot would leave the room and the widget would break. It felt easier to just have the bot stay in the room, and have the user kick the bot if it was taking up valuable membership space.